pinot-query-planner/src/test/java/org/apache/pinot/query/testutils/QueryTestUtils.java

```diff
@@ -0,0 +1,40 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.pinot.query.testutils;
+
+import java.io.IOException;
+import java.net.ServerSocket;
+
+
+public class QueryTestUtils {
+
+  private QueryTestUtils() {
+    // do not instantiate.
+  }
+
+  public static int getAvailablePort() {
```

Comment:
this should be `synchronized` (surprisingly, I've hit issues with this in a previous life)

Reply:
fixed this one, but also added a java doc explaining this is still not threadsafe/potential race condition can still occur